core/azure-core-amqp/src/main/java/com/azure/core/amqp/exception/ExceptionUtil.java

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.exception;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class to map AMQP status codes and error conditions to an exception.
+ */
+public final class ExceptionUtil {
```

Comment:
Is this class intended for public API or just your internal use? I suspect public API but I want to double-check.

Reply:
It's for public use. Makes it easy to translate AMQP errorCondition strings to an actual Exception. Is there a better way to put it? Like some converter class?